rudder-core/src/main/scala/com/normation/rudder/services/policies/write/Cf3PromisesFileWriterService.scala

```diff
@@ -0,0 +1,620 @@
+import com.normation.cfclerk.domain.SystemVariableSpec
+import com.normation.cfclerk.domain.Variable
+import com.normation.cfclerk.domain.TrackerVariableSpec
+import com.normation.inventory.domain.AgentType
+import com.normation.inventory.domain.NodeId
+import com.normation.inventory.domain.{COMMUNITY_AGENT, NOVA_AGENT}
+import com.normation.rudder.domain.reports.NodeConfigId
+import com.normation.rudder.repository.LicenseRepository
+import com.normation.rudder.services.nodes.NodeInfoService
+import com.normation.rudder.services.policies.SystemVariableService
+import com.normation.rudder.services.policies.nodeconfig.NodeConfiguration
+import com.normation.rudder.services.reports.ReportingService
+import com.normation.stringtemplate.language.NormationAmpersandTemplateLexer
+import com.normation.stringtemplate.language.formatter.LocalTimeRenderer
+import com.normation.stringtemplate.language.formatter.DateRenderer
+import com.normation.stringtemplate.language.formatter.LocalDateRenderer
+import com.normation.utils.Control.sequence
+import com.normation.utils.Control.boxSequence
+import org.antlr.stringtemplate.StringTemplate
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.IOUtils
+import org.apache.commons.io.FilenameUtils
+import org.joda.time.LocalTime
+import org.joda.time.LocalDate
+import org.joda.time.DateTime
+import net.liftweb.util.Helpers.tryo
+import net.liftweb.common.Full
+import net.liftweb.common.Box
+import net.liftweb.common.Failure
+import net.liftweb.common.Loggable
+import net.liftweb.common.EmptyBox
+import com.normation.rudder.services.policies.BundleOrder
+import com.normation.rudder.services.policies.nodeconfig.NodeConfigurationLogger
+
+
+/**
+ * Write promises for the set of nodes, with the given configs.
+ * Requires access to external templates files.
+ */
+trait Cf3PromisesFileWriterService {
+
+  /**
+   * Write templates for node configuration that changed since the last write.
+   *
+   */
+  def writeTemplate(rootNodeId: NodeId, nodesToWrite: Set[NodeId], allNodeConfigs: Map[NodeId, NodeConfiguration], versions: Map[NodeId, NodeConfigId]) : Box[Seq[NodeConfiguration]]
+}
+
+
+
+class Cf3PromisesFileWriterServiceImpl(
+    techniqueRepository      : TechniqueRepository
+  , pathComputer             : PathComputer
+  , licenseRepository        : LicenseRepository
+  , logNodeConfig            : NodeConfigurationLogger
+  , prepareTemplate          : PrepareTemplateVariables
+  , communityCheckPromises   : String
+  , novaCheckPromises        : String
+  , cfengineReloadPromises   : String
+) extends Cf3PromisesFileWriterService with Loggable {
+
+
+  /**
+   * Write templates for node configuration that changed since the last write.
+   *
+   */
+  override def writeTemplate(rootNodeId: NodeId, nodesToWrite: Set[NodeId], allNodeConfigs: Map[NodeId, NodeConfiguration], versions: Map[NodeId, NodeConfigId]) : Box[Seq[NodeConfiguration]] = {
```

Comment:
could you change the signature so that each entries are on one line like every other methods ?